uv:
  #Publish all joint state
  uv_joint_state_controller:
    type: "joint_state_controller/JointStateController"
    publish_rate: 30

  #Differential drive controller
  uv_diffdrive_controller:
    type: "diff_drive_controller/DiffDriveController"
    publish_rate: 30
    left_wheel: uv_left_wheel_joint
    right_wheel: uv_right_wheel_joint
    wheel_radius: 0.11

    pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]
    twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]
    # Top level frame (link) of the robot description
    base_frame_id: uv_base_footprint
    odom_frame_id: uv_odom
    # Velocity and acceleration limits for the robot
    linear:
      x:
        has_velocity_limits    : true
        max_velocity           : 2.5   # m/s
        has_acceleration_limits: true
        max_acceleration       : 1.0  # m/s^2
    angular:
      z:
        has_velocity_limits    : true
        max_velocity           : 2.5  # rad/s
        has_acceleration_limits: true
        max_acceleration       : 2.5  # rad/s^2

drug:
  drug_joint_state_controller:
    type: "joint_state_controller/JointStateController"
    publish_rate: 30  

  drug_diffdrive_controller:
    type: "diff_drive_controller/DiffDriveController"
    publish_rate: 30
    left_wheel: drug_left_wheel_joint
    right_wheel: drug_right_wheel_joint
    wheel_radius: 0.11

    pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]
    twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]
    # Top level frame (link) of the robot description
    base_frame_id: drug_base_footprint
    odom_frame_id: drug_odom
    # Velocity and acceleration limits for the robot
    linear:
      x:
        has_velocity_limits    : true
        max_velocity           : 2.5   # m/s
        has_acceleration_limits: true
        max_acceleration       : 1.0  # m/s^2
    angular:
      z:
        has_velocity_limits    : true
        max_velocity           : 2.5  # rad/s
        has_acceleration_limits: true
        max_acceleration       : 2.5  # rad/s^2

drug1:
  drug1_joint_state_controller:
    type: "joint_state_controller/JointStateController"
    publish_rate: 30  

  drug1_diffdrive_controller:
    type: "diff_drive_controller/DiffDriveController"
    publish_rate: 30
    left_wheel: drug1_left_wheel_joint
    right_wheel: drug1_right_wheel_joint
    wheel_radius: 0.11

    pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]
    twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.01]
    # Top level frame (link) of the robot description
    base_frame_id: drug1_base_footprint
    odom_frame_id: drug1_odom
    # Velocity and acceleration limits for the robot
    linear:
      x:
        has_velocity_limits    : true
        max_velocity           : 2.5   # m/s
        has_acceleration_limits: true
        max_acceleration       : 1.0  # m/s^2
    angular:
      z:
        has_velocity_limits    : true
        max_velocity           : 2.5  # rad/s
        has_acceleration_limits: true
        max_acceleration       : 2.5  # rad/s^2